### DISCUSSION TITLE
How use E3V2-UBL-BLTouch-10x10-HS-v4.2.2.bin

### DISCUSSION THREAD

**Discussion by MrDragon420** [Jyers/Marlin#342] [Q&A] [answered]
What are the proper steps for useing / doing UBL 
I have offsets -48 for x and - 10 for Y

**Comment by Jyers** [comment, marked as answer]
I will hopefully have a guide/video on this soon as it seems to be a common confusion. The general procedure is to select create mesh to make a new mesh, save it, and then select tilt mesh whenever you want to correct for slight bed movements. Here's a quick rundown on the gcode and features https://github.com/Jyers/Marlin/wiki/Configuration-Details

**Comment by smooreace** [comment]
Funny... was looking for some form of validation for my offset measurements for a BLT at  -44.62, -9.17

I guess I wasn't totally screwing up. Just wish there was a good video that covered everything for my first delve into all of this. I tried the pre-compiled file and it refuses to boot now

**Reply by Jyers** [reply]
Which precompiled were you using? Failure to boot is often caused by the sd card, so you may want to try reformatting it.

**Reply by smooreace** [reply]
I actually had 2 mistakes... I didn't confirm my board. Thought I was running a 4.2.7 but in fact had a 4.2.2. Also, I was trying to flash from a 2 and 4 GB SD card. Once I changed to an 8GB card the flash went fine. The 4.2.7 flash installed but gave no stepper control, which triggered the thought to confirm my hardware.

**Reply by smooreace** [reply]
I thought I had everything working right. Z offset is set as well. I added G29 to my start g-code. But when I start a print job it does its bed scan and then sits in the back right corner, drops the head down to print height and dumps filament in a puddle.... infinitely. Its not even trying to move.

**Reply by Jyers** [reply]
Sounds like a gcode error of some kind. Have you printed the file before?

**Reply by smooreace** [reply]
Printed a simple bed leveling pattern that was pre-sliced and still everything was in the back right corner. This would normally take up the entire bed and printed the bottom left 1/3rd of the print in the top right 1/3rd of the bed. So it's an offset issue I guess. But somehow the bed leveling is working properly

**Reply by smooreace** [reply]
I'm going to put stock Creality with BLT on the thing and see if that solves it. Right now Im using your pre-compiled 
E3V2-BLTouch-5x5-HS-v4.2.2.bin

**Reply by smooreace** [reply]
Stock Creality firmware is doing the same thing

**Reply by Jyers** [reply]
After flashing, make sure you reset eeprom as that can sometimes cause issues.

**Reply by smooreace** [reply]
Reflashed... again, using your E3V2-BLTouch-3x3-HS-v4.2.2.bin
Loaded default settings
Added my probe offsets of -43.6,-9.2,-2.44
Saved Settings
Ran Auto Level 
Saved
Auto Home brought me to 153.6,119.2,12.44

Is home supposed to be near the center of the bed? That could explain why its only printing in the top right corner?

**Reply by smooreace** [reply]
I also tried compiling my own bin with no changes and it fails to compile. Was attempting to force home to 0,0 + offsets but when it wouldn't compile as a fresh load I didn't move forward

**Reply by Jyers** [reply]
To compile your own, you'll want to copy the config files for your setup from the configuration templates folder into the marlin folder.

**Reply by Jyers** [reply]
But if it is happening with stock firmware as well, it would seem to be a hardware or gcode issue

**Reply by smooreace** [reply]
This is the tail end of a G29 leveling added to the start g-code
https://youtu.be/kDDNGwJDBAY
And the result
![20210403_130251](https://user-images.githubusercontent.com/35540547/113487422-ac736a80-947d-11eb-99ee-2ff770248749.jpg)

**Reply by Jyers** [reply]
Interesting, perhaps your missing the part of your start gcode that puts the printer back into absolute positioning. Is there a G90 in there somewhere? If thats not the case perhaps posting your entire start gcode could help to figure out the issue.

**Reply by smooreace** [reply]
One problem was that I didnt tell Cura that **Home** was in the center

Fixed that and reprinted and got
![20210403_132249](https://user-images.githubusercontent.com/35540547/113488002-f0b43a00-9480-11eb-9885-e9203b3576b8.jpg)

So we are in a LOT better shape now. Its printing 40.6,5.65 off from center.


Gcode start...
;FLAVOR:Marlin
;TIME:338
;Filament used: 0.203116m
;Layer height: 0.2
;MINX:-111
;MINY:-111
;MINZ:0.2
;MAXX:111
;MAXY:111
;MAXZ:0.2
;Generated with Cura_SteamEngine 4.8.0
M140 S75
M105
M190 S75
M104 S210
M105
M109 S210
M82 ;absolute extrusion mode
G28 ; home all axes
G92 X-5.0 Y-10.0 ; set home offsets
G1 X0 Y0 Z0.2 F3000 ; get ready to prime
G92 E0 ; reset extrusion distance
G1 X160 E15 F600 ; prime nozzle
G1 X180 F5000 ; quick wipe
G92 E0
G92 E0
G1 F1500 E-1.5

**Reply by Jyers** [reply]
Why are you telling Cura home is in the center? On an E3V2 home should be in the front left corner, where the endstops are

**Reply by Jyers** [reply]
`G92 X-5.0 Y-10.0 ; set home offsets` this is your issue. With a BL Touch the head has to go to the middle of the bed to probe, after which you immediately set that point to -5,-10 with that G92. Either remove that line or use M206, which is the command that actually sets your home offsets rather than just setting the current position.
